src/magic_folder/tahoe_client.py

```diff
@@ -335,22 +367,29 @@ def add_entry_to_mutable_directory(self, mutable_cap, path_name, entry_cap, repl
         returnValue(capability_string)
 
     @inlineCallbacks
-    def download_capability(self, cap):
+    def download_file(self, cap):
         """
-        Retrieve the raw data for a capability from Tahoe
+        Retrieve the raw data for a capability from Tahoe. It is an error
+        if the capability-string is a directory-capability, since the
+        Tahoe `/uri` endpoint treats those specially.
 
         :param cap: a capability-string
 
         :returns: bytes
         """
         # Visiting /uri with a directory-capability causes Tahoe to
         # render an HTML page .. instead adding ?t=json tells it to
-        # send JSON instead. However, using ?t=json for non-directory
-        # capabilities means the content is *not* downloaded .. so,
-        # "t=json" is only added for directory-capabilities.
-        query_args = [(u"uri", cap.decode("ascii"))]
+        # send the JSON instead. We can't just use ?t=json all the
+        # time though, because for non-directories it returns the
+        # metadata, not the data. So, the caller needs to know if they
+        # have a dir-cap or not and call list_directory() or
+        # directory_data() to get "raw" representation
         if is_directory_cap(cap):
```

Comment:
Instead of checking that it isn't a directory cap, can you check that it is a file cap? I think this means `IFileURI` or `IMutableFileURI`. After making this suggestion, I see that this avoid `IVerifierURI`, but I think generally allow-lists make more sense than deny-lists for this kind of check.

Reply:
Strictly speaking, Tahoe could have new as-yet-unknown URIs that would download just fine (remember the point of this restriction/change is that the Tahoe endpoint renders HTML for dir-caps, but not for other things) .. 

.. but, okay, we say "file" in the API now so I suppose it makes sense to insist it's only one of the known file-cap types.